.github/workflows/sphinx.yml

```diff
@@ -0,0 +1,38 @@
+name: build-sphinx-to-gh-pages
+on:
+  push:
+    branches:
+    - master
+
+jobs:
+  build_sphinx_job:
+    runs-on: ubuntu-latest
+    steps:
+
+      - name: Set up Python 3.9
+        uses: actions/setup-python@v1
+        with:
+          python-version: 3.9
+
+      - name: Install apt dependencies
+        run: |
+          sudo apt-get install -y libhdf5-serial-dev libopenmpi-dev openmpi-bin
+      - uses: actions/checkout@v2
+      - name: Install pip requirements
+        run: |
+          pip install -r requirements.txt
+          pip install -r doc/requirements.txt
+          pip install .
+      - name: Sphinx build
+        run: |
+          git checkout -b gh-pages
+          mkdir docs
+          sphinx-build -b html doc/ docs
+          touch docs/.nojekyll
+      - name: Commit site
+        run: |
+          git config --global user.name 'Rick Nitsche (automated)'
+          git config --global user.email 'nritsche@users.noreply.github.com'
+          git add .
+          git commit -am "Automated sphinx build"
+          git push --force origin gh-pages
```

Comment:
Why `--force`?

Reply:
Good question. Mostly because this is copypasta. I also questioned this, but concluded: "why not?" this is committing the compiled html docs to a special branch. If it builds, we always want to push it. Maybe there could be 2 instances of this running at the same time, and we don't want the CI to fail because of that (even though the docs wouldn't be complete after they ran both at the same time?). I could be convinced to remove it...

Reply:
That makes sense. I do not feel strongly about removing it.